### PR TITLE
feat: codux migration

### PR DESCRIPTION
**Pull request by TheAlmightyCrumb**
Following recent changes moving from WCS to Codux, a migration done according to [Boilerplate Structures](https://docs.google.com/document/d/1TroPGXxQYTDQeVUFC7LfVZu0iwZHgQegRPATTWOhsBo/edit#)
You can see the changes and commits: https://github.com/wixplosives/blank-scss/commits/sagivd/codux-migration

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+import logo from "./assets/logo.svg";
+import styles from "./App.module.scss";
 
 function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
+    <div className={styles.App}>
+      <header className={styles["App-header"]}>
```
Comment (YardenPorat) [inline src/App.tsx:8]: when using modules solutions, there is a convention to use camelCase for class names.

File: src/App.tsx [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
+        <img src={logo} className={styles["App-logo"]} alt="logo" />
```
Comment (YardenPorat) [inline src/App.tsx:9]: same

File: package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "@types/node": "^18.11.9",
     "@types/react": "^18.0.25",
     "@types/react-dom": "^18.0.9",
+    "classnames": "^2.3.2",
```
Comment (YardenPorat) [inline package.json:13]: why?

Comment (YardenPorat) [reply package.json:13]: not in line with PR description

File: src/App.tsx [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
         <p>
           Edit <code>src/App.tsx</code> and save to reload.
         </p>
         <a
-          className="App-link"
+          className={styles["App-link"]}
```
Comment (YardenPorat) [inline src/App.tsx:14]: same

**Review by YardenPorat — COMMENTED** [review]
[src/assets/logo.svg](https://github.com/studio-templates/blank-scss/pull/9/files#diff-8700fcf7731563aa80c6e3a7589f00e228da89bb7535307f3f85f9c553466d35)

Why are we changing structure? This was supposed to follow `create-react-app` structure. was that decision changed? where is it documented?

File: tsconfig.json [hunk 5]
```diff
@@ -1,26 +1,23 @@
...
+        "forceConsistentCasingInFileNames": true,
+        "strict": true,
+        "noImplicitReturns": true,
+        "noFallthroughCasesInSwitch": true,
+        "noUncheckedIndexedAccess": true,
+        "noImplicitOverride": true,
+        "skipDefaultLibCheck": true,
+        "noEmit": true
+    }
 }
```
Comment (YardenPorat) [inline tsconfig.json:23]: why

**Review by YardenPorat — CHANGES_REQUESTED** [review]
🦖

File: tsconfig.json (hunk 5 above)
Comment (TheAlmightyCrumb) [reply tsconfig.json:23]: From the doc: [`tsconfig.json`](https://docs.google.com/document/d/1TroPGXxQYTDQeVUFC7LfVZu0iwZHgQegRPATTWOhsBo/edit#heading=h.c8yb07nmvpow)

**Pull request closed without merging**